dbtvault-dev/macros/tables/sat.sql

```diff
@@ -62,8 +64,13 @@ records_to_insert AS (
     {%- if dbtvault.is_any_incremental() %}
     LEFT JOIN latest_records
     ON {{ dbtvault.prefix([src_pk], 'latest_records', alias_target='target') }} = {{ dbtvault.prefix([src_pk], 'stage') }}
+    {% if target.type == 'snowflake' %}
     WHERE {{ dbtvault.prefix([src_hashdiff], 'latest_records', alias_target='target') }} != {{ dbtvault.prefix([src_hashdiff], 'stage') }}
         OR {{ dbtvault.prefix([src_hashdiff], 'latest_records', alias_target='target') }} IS NULL
+    {% elif target.type == 'bigquery' %}
+    WHERE {{ dbtvault.prefix([src_hashdiff], 'latest_records', alias_target='target') }} != {{ dbtvault.prefix([src_hashdiff], 'stage') }}
+        OR {{ dbtvault.prefix([src_hashdiff], 'latest_records', alias_target='target') }} IS NULL
```

Comment:
Ensure you have a filter here for rank = 1 to replace QUALIFY

Reply:
Not sure if this is relevant, but bigquery supports QUALIFY as well.

The only strange caveat is that there must be at least 1 expression in the WHERE clause of the same query for it to work (so I sometimes add `WHERE 1 = 1`, when there is no filter)

Reply:
Thanks for this! We are planning to go through a re-factor phase before we release this publicly and will be sure to look in to these kinds of things.